bound/bound.go

```diff
@@ -4,7 +4,37 @@
 
 package bound
 
+import "math"
+
 // Bound represents [Min, Max] bounds.
 type Bound struct {
 	Min, Max float64
 }
+
+// Intersection returns a intersected bound between bounds.
+// if len(bounds) == 0 then it will return NaN value bound.
+// Or if there is no intersection bound then it will return Nan value bound.
+func Intersection(bounds ...Bound) Bound {
+	if len(bounds) == 0 {
+		return Bound{Min: math.NaN(), Max: math.NaN()}
```

Comment:
If NaN is being used we should have a helper to test whether the `Bound` is valid.

Reply:
BTW Is there a reason to not just use the zero value? It is an empty set and so expresses what this would.

Reply:
@kortschak I would like to distinguish between empty bound and intersected bound which actually has zero value.

I 've updated it to return an empty value.

Reply:
We have a problem here that stems from the use of closed intervals for `Bound`. My comment above is not correct, [x, x] is not an empty set, and since [0, 0] is both in that set of sets, and 0 is a valid number, the zero value for `Bound` can't be used as a semaphore. So this answers my question.

This means that we *do* need to use NaN, but we should also have a way to test validity. Even if only to signal to users that `b1 == b2` may not necessarily be meaningful for all `Bound`s.

Reply:
Names for the helper (I don't have firm favourites): `IsEmpty`, `IsValid`, `IsZero`. Obviously the semantics of `IsEmpty`/`IsZero` and `IsValid` are inverted. (I think I prefer `IsEmpty(b Bound) bool { return math.IsNaN(b.Min) || math.IsNaN(b.Max) }`.